samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -279,10 +287,10 @@ def function_name(self):
 
         all_functions = list(self._function_provider.get_all())
         if len(all_functions) == 1:
-            return all_functions[0].functionname
+            return all_functions[0].name
```

Comment:
Why we are changing from `functionname` to `name`?

Reply:
The explanation is hidden in the commit message of f4f767d6a786e4b20705df85c24e13754abb608a:

> change the return value when user did not provide an identifier
from .functionname to .name. This is to make it consistent with
LocalLambda.invoke(), where in line 104
`all_functions = [f.name for f in self.provider.get_all()]`

Reply:
As discussed offline, we also changed the error message that list all available options to use full_path instead of just the function name